### DISCUSSION TITLE
Actions with cherry pick from bash script

### DISCUSSION THREAD

**Discussion by andrewmitchellsemestry** [community/community#33153] [Actions] [answered]
Git Action not carrying out git cherry-pick.

We are currently trying to implement a testing and staging branch (just like master and production) so that we can test our code changes independently before adding them to the production branch.

We have setup a github action that works in tandem with a bash script to do this. The bash script is quite simple: it looks for all commit hashes related to the jira that the work was carried out on, creates a new branch on the staging branch to house these commits and cherry picks each hash so the we can create a PR with the newly created branch for review.

The problem we are facing is that the cherry-pick will never work and it always causes a merge conflict, even when the change is as simple as removing one line of whitespace.

When running the bash script locally, the new branch is created without any issues.

Does anyone know of any reasons why the environment on which the github action is run would produce these conflicts every time?
![image](https://user-images.githubusercontent.com/78592641/190399656-c03d4799-2e66-4adc-951c-f7d7c692ca65.png)

**Comment by airtower-luna** [comment, marked as answer]
Exactly how (error messages, etc.) does the script not work? From looking at your workflow I'd expect the script call to fail because the script doesn't exist. You need to [check out the repository](https://github.com/actions/checkout) before you can call a script stored in your repository.

**Reply by andrewmitchellsemestry** [reply]
Apologies, I had uploaded the incorrect screenshot (updated now). The first step is to checkout the repository.

The script is "failing" due to the commits not being cherry picked.

![image](https://user-images.githubusercontent.com/78592641/190400378-7f0a31d4-d642-4d25-814a-001abdc9b103.png)

 We added some more logging in (git log -n 5 --oneline) and it looks as though the workspace only has reference to the last commit and none before this. Would this be the case where the whole git history is not considered?

**Reply by airtower-luna** [reply]
That actually makes sense, by default `actions/checkout` gets only the latest commit. You can get the whole history by setting `fetch-depth` to 0:

```yaml
- uses: actions/checkout@v3
  with:
    fetch-depth: 0
```

**Reply by andrewmitchellsemestry** [reply]
Brilliant, that is exactly what we needed! Thanks for that

**Reply by airtower-luna** [reply]
Glad I could help! Marking the answer would be appreciated. :slightly_smiling_face:
